lib/evm_database_ops.rb

```diff
@@ -89,7 +89,8 @@ def self.restore(db_opts, connect_opts = {})
     #   :username => 'samba_one',
     #   :password => 'Zug-drep5s',
 
-    uri = with_mount_session(:restore, db_opts, connect_opts) do |database_opts, _session, _remote_file_uri|
+    uri = with_mount_session(:restore, db_opts, connect_opts) do |database_opts, session, remote_file_uri|
+      database_opts[:local_file] = session.download(database_opts[:local_file], remote_file_uri) if session
```

Comment:
So I'm thinking that this should only reassign `database_opts[:local_file]` if the local file doesn't exist already.

I'm worried that we're clobbering the value set on line 122 for mount types that actually mount.

Maybe something like this:
```ruby
database_opts[:local_file] = session&.download(database_opts[:local_file], remote_file_uri) unless File.exist?(database_opts[:local_file])
```

What do you think?

Reply:
I feel like `database_opts[:local_file]` will be set for the S3 case, but won't actually exist yet because we don't mount, but for the other cases we should just restore from the mounted filesystem rather than essentially re-running the `download`.